js/components/App.js

```diff
@@ -37,43 +38,110 @@ import DashTabs from "./DashTabs";
 /* The maximum number of letter examples to load (and possibly show) */
 const MAX_EXAMPLES = 5;
 
-const API_ROOT = "https://db.syriac.reclaim.hosting/api/";
+export const API_ROOT = "https://db.syriac.reclaim.hosting/api/";
 //const API_ROOT = "http://localhost:8000/api/";
 
 class App extends Component {
   constructor(props) {
     super(props);
 
     this.state = {
-      manuscripts: [],
-      allManuscripts: [],
+      manuscripts: [], // Only the selected mss, sent to the table (sorted)
+      allManuscripts: [], // Populates the form list, to be sorted as needed
       showTabs: false,
       formData: {},
-      tableData: {}
+      tableData: {},
+      loadingMessage:
+        'Please select one or more manuscripts and letters from the options menu, then click the "Submit" button.'
     };
 
-    this.queryCache = {};
+    //this.queryCache = {}; // XXX Just let the browser cache handle this?
 
     this.handleSubmit = this.handleSubmit.bind(this);
     this.queryManuscripts = this.queryManuscripts.bind(this);
     this.processCoords = this.processCoords.bind(this);
+    this.getYearFromDate = this.getYearFromDate.bind(this);
+    this.sortManuscripts = this.sortManuscripts.bind(this);
+  }
+
+  /* The ms date field in the DB is a bit unruly. We'll do our best
+   * to parse it to a sortable 4-digit year, or else return 'NA'.
+   */
+  getYearFromDate(date) {
```

Comment:
Do we imagine needing this elsewhere? If so, we could pull this out into a utility function file, along with some other functions if we find them. Could also do that just to keep the components smaller. It's an optional change to this PR.

Reply:
I haven't anticipated another case in which it will be needed, and at the moment it's more or less on its own as a utility function. But if it needs to be used in another component, or if another utility function joins the team, then it definitely should go into a separate file.